Azure/azure-cli
# [Backup] `az backup vault create`: Add new parameter `--immutability-state` to support configuring immutability settings for the backup vault

**Related command**


`az backup vault create`

**Description**

(Tested using the recoveryservices SDK version 2022-10-01) The PR adds support for the updated recoveryservices SDK that specifies an `immutability-state` flag for vaults, impacting vault creation/editing/display. Vault display (`az backup vault show`) is handled by the existing code, we need to modify behaviour of vault create/delete (Both handled by `az backup vault create`).

**Testing Guide**


`az backup vault show`

When querying for a vault which has immutability-state set to any of the three values, under "properties" we should see:

```
"securitySettings": {
    "immutabilitySettings": {
        "state": "<Value>"
    },
    "multiUserAuthorization": "Invalid",
    "softDeleteSettings": null
},
```

`az backup vault create`

When creating a new vault, we can set the flag manually: if we set it to "Disabled", the result will be "immutabilitySettings.state: Disabled", and likewise for "Unlocked" or "Locked".
This can also be used to edit vaults.

**History Notes**


---

This checklist is used to make sure that common guidelines for a pull request are followed.
